spring-kafka/src/test/java/com/baeldung/partitioningstrategy/KafkaApplicationUnitTesting.java

```diff
@@ -0,0 +1,164 @@
+package com.baeldung.partitioningstrategy;
+
+import static org.junit.Assert.assertEquals;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.junit.Assert;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.kafka.core.DefaultKafkaProducerFactory;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.test.EmbeddedKafkaBroker;
+import org.springframework.kafka.test.context.EmbeddedKafka;
+import org.springframework.kafka.test.utils.KafkaTestUtils;
+
+@SpringBootTest
+@EmbeddedKafka(partitions = 3, brokerProperties = {"listeners=PLAINTEXT://localhost:9092"})
+public class KafkaApplicationUnitTesting {
+    @Autowired
+    private KafkaProducer kafkaProducer;
+
+    @Autowired
+    private KafkaMessageConsumer kafkaMessageConsumer;
+
+    @Autowired
+    private EmbeddedKafkaBroker embeddedKafkaBroker;
+
+    @Autowired
+    private Consumer<String, String> consumer;
+
+    @BeforeEach
+    public void clearMessages() {
+        kafkaMessageConsumer.clearReceivedMessages();
+    }
+
+    @Test
+    public void givenDefaultPartitioner_whenSendingMessagesWithoutKey_shouldUseStickyDistribution() throws InterruptedException {
+        kafkaProducer.send("default-topic", "message1");
+        kafkaProducer.send("default-topic","message2");
+        kafkaProducer.send("default-topic","message3");
+
+        Thread.sleep(2000);
+
+        List<ReceivedMessage> records = kafkaMessageConsumer.getReceivedMessages();
+        int expectedPartition = records.get(0).getPartition();
+
+        for (ReceivedMessage record : records) {
+            if (record.getKey() == null) {
+                Assert.assertEquals("Message without key should be in the same partition", expectedPartition, record.getPartition());
+            }
+        }
```

Comment:
What is the purpose of the if statement here? Isn't every message sent without a key in this test?